File: leetcode/practice/1323-maximum-69-number/without-conversion.py
class Solution:
    def maximum69Number (self, num: int) -> int:
        temp = num
        ind , i = -1 , 0
        while temp != 0 :
            ind = i if temp%10 == 6 else ind
            temp //= 10
            i += 1
        return (num + 3*(10**ind) if ind != -1 else num)

File: leetcode/practice/1323-maximum-69-number/without-conversion.cpp
class Solution {
public:
    int maximum69Number (int num) ;
};

int Solution :: maximum69Number(int num)
{
    int temp = num ;
    int ind = -1 , i = 0 ;
    while(temp != 0)
    {
        ind = temp%10 == 6 ? i : ind ;
        temp /= 10 ;   
        i++ ;   
    }
    return num + 3*(ind != -1 ? pow(10 , ind) : 0) ;
}
